src/main/java/graphql/schema/idl/SchemaPrinter.java

```diff
@@ -819,21 +850,25 @@ private SchemaElementPrinter<GraphQLSchema> schemaPrinter() {
             }
 
             if (needsSchemaPrinted) {
-                if (hasAstDefinitionComments(schema) || hasDescription(schema)) {
-                    out.print(printComments(schema, ""));
-                }
-                List<GraphQLAppliedDirective> directives = DirectivesUtil.toAppliedDirectives(schema.getSchemaAppliedDirectives(), schema.getSchemaDirectives());
-                out.format("schema %s{\n", directivesString(GraphQLSchemaElement.class, directives));
-                if (queryType != null) {
-                    out.format("  query: %s\n", queryType.getName());
-                }
-                if (mutationType != null) {
-                    out.format("  mutation: %s\n", mutationType.getName());
-                }
-                if (subscriptionType != null) {
-                    out.format("  subscription: %s\n", subscriptionType.getName());
+                if (shouldPrintAsAst(schema.getDefinition())) {
+                    printAsAst(out, schema.getDefinition(), schema.getExtensionDefinitions());
```

Comment:
Main change on this line: if AST flag is on, use the new AST print method for SchemaDefinition